tests/phpunit/tests/dependencies.php

```diff
@@ -148,4 +148,88 @@ public function test_enqueue_before_register() {
 
 		$this->assertContains( 'one', $dep->queue );
 	}
+
+	/**
+	 * Data provider for test_get_etag.
+	 *
+	 * @return array
+	 */
+	public function data_provider_get_etag() {
+		return array(
+			'should accept one dependency'              => array(
+				'load'               => array(
+					'abcd' => '1.0.2',
+				),
+				'hash_source_string' => 'WP:abcd:1.0.2;',
+				'expected'           => 'W/"fea2fd8012dd8af0696ebafbaa68db85"',
+			),
+			'should accept empty array of dependencies' => array(
+				'load'               => array(),
+				'hash_source_string' => 'WP:;',
+				'expected'           => 'W/"f6457280f73cc597e76df87ee891457a"',
+			),
+			'should accept more then one dependency and wp version' => array(
+				'load'               => array(
+					'abcd' => '1.0.2',
+					'abdy' => '1.0.3',
+				),
+				'hash_source_string' => 'WP:5.4.3;abcd:1.0.2;abdy:1.0.3;',
+				'expected'           => 'W/"88649143b0142d1491883065e9351178"',
+			),
+		);
+	}
+
+	/**
+	 * Tests get_etag method for WP_Scripts.
+	 *
+	 * @ticket 58433
+	 * @ticket 61485
+	 *
+	 * @covers WP_Dependencies::get_etag
+	 *
+	 * @dataProvider data_provider_get_etag
+	 *
+	 * @param array $load List of scripts to load.
+	 * @param string $hash_source_string Hash source string.
+	 * @param string $expected Expected etag.
+	 */
+	public function test_get_etag_scripts( $load, $hash_source_string, $expected ) {
+		$instance = wp_scripts();
+
+		foreach ( $load as $handle => $ver ) {
+			// The src should not be empty.
+			wp_enqueue_script( $handle, 'https://example.cdn', array(), $ver );
```

Comment:
Use example.org as it's never going to be a site. example.cdn may eventually become a legitimate domain containing content outside of our control.

```suggestion
			wp_enqueue_script( $handle, 'https://example.org', array(), $ver );
```